lib/online_order.rb

```diff
@@ -0,0 +1,128 @@
+require_relative './order.rb'
+
+module Grocery
+
+  class OnlineOrder < Order
+    attr_reader :customer_id, :status
+
+    def initialize (id, products, customer_id, status)
+      super(id, products)
+      # @online_total = online_total
+      @status = status
+      @customer_id = customer_id
+    end
+
+    def total
+      return 10 + super
+    end
+
+    def self.find(id)
+      online_orders = OnlineOrder.all
+      online_orders.each do |o|
+        if o.id == id
+          return o
+        end
+      end
+
+      return nil
+    end
+
+    def self.all
+      require 'csv'
+      # csv = CSV.open("../support/online_orders.csv")r
+      csv = CSV.open("support/online_orders.csv")
+      online_orders_list = []
+      csv.each do |l|
+        products = { }
+        #puts l[1].split(';')
+
+        product_string = l[1]
+        customer_id = l[2].to_i
+        status = l[3]
+
+        product_smash = product_string.split(';')
+        product_smash.each do |ps|
+          # ps = "name1:val1"
+          items = ps.split(':')
+          # hash_entry = {items[0] => items[1].to_f}
+          # products.push(hash_entry)
+          products[items[0]] = items[1].to_f
+
+        end  # end of product_smash.each
+
+        online_order = OnlineOrder.new(l[0].to_i, products,customer_id,status)
+        online_orders_list.push(online_order)
+
+      end # end of csv.each
+
+      return online_orders_list
+    end
+
+    def self.find_by_customer(customer_id)
```

Comment:
This method should return an **Array** of `OnlineOrder` objects not just the first one that has that customer.  

So... not quite right.